enterprise/server/remote_execution/runner/runner.go

```diff
@@ -1212,29 +1211,33 @@ func (r *CommandRunner) sendPersistentWorkRequest(ctx context.Context, command *
 func (r *CommandRunner) marshalWorkRequest(requestProto *wkpb.WorkRequest, writer io.Writer) error {
 	protocol := r.PlatformProperties.PersistentWorkerProtocol
 	if protocol == workerProtocolJSONValue {
-		marshaler := jsonpb.Marshaler{EmitDefaults: true}
-		if err := marshaler.Marshal(writer, requestProto); err != nil {
+		marshaler := &protojson.MarshalOptions{EmitUnpopulated: true}
+		out, err := marshaler.Marshal(requestProto)
+		if err != nil {
 			return err
 		}
-		_, err := fmt.Fprintf(writer, "\n")
+		_, err = fmt.Fprintf(writer, "%s\n", string(out))
 		return err
 	}
 	if protocol != "" && protocol != workerProtocolProtobufValue {
 		return status.FailedPreconditionErrorf("unsupported persistent worker type %s", protocol)
 	}
-	buf := proto.NewBuffer( /* buf */ nil)
-	if err := buf.EncodeMessage(requestProto); err != nil {
+	out, err := proto.Marshal(requestProto)
+	if err != nil {
 		return err
 	}
-	_, err := writer.Write(buf.Bytes())
+	_, err = writer.Write(out)
 	return err
 }
 
 func (r *CommandRunner) unmarshalWorkResponse(responseProto *wkpb.WorkResponse, reader io.Reader) error {
 	protocol := r.PlatformProperties.PersistentWorkerProtocol
 	if protocol == workerProtocolJSONValue {
-		unmarshaller := jsonpb.Unmarshaler{AllowUnknownFields: true}
-		return unmarshaller.UnmarshalNext(r.jsonDecoder, responseProto)
+		raw := json.RawMessage{}
+		if err := r.jsonDecoder.Decode(&raw); err != nil {
+			return err
+		}
+		return (&protojson.UnmarshalOptions{DiscardUnknown: true}).Unmarshal(raw, responseProto)
```

Comment:
it looks like Unmarshal is defined on the UnmarshalOptions struct directly (no pointer receiver), so I think you should be able to change all the callsites like:

(&protojson.UnmarshalOptions{}).Unmarshal(blah) to be:
protojson.UnmarshalOptions{}.Unmarshal(blah) ?

Reply:
Done!